src/main/java/seedu/foodrem/model/item/TagSetContainsTagPredicate.java

```diff
@@ -0,0 +1,38 @@
+package seedu.foodrem.model.item;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.foodrem.model.tag.Tag;
+
+/**
+ * Tests that a {@code Item}'s {@code TagSet} contains the specified {@code tag}
+ */
+public class TagSetContainsTagPredicate implements Predicate<Item> {
+    private final Tag tag;
+
+    /**
+     * @param tag Checks if tag exists in the item's tagSet
+     */
+    public TagSetContainsTagPredicate(Tag tag) {
+        requireNonNull(tag);
+        this.tag = tag;
+    }
+
+    @Override
+    public boolean test(Item item) {
+        return item.getTagSet().contains(tag);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this
+                || (other instanceof TagSetContainsTagPredicate
+                && tag.equals(((TagSetContainsTagPredicate) other).tag));
+    }
+
+    public Tag getTag() {
+        return this.tag;
+    }
```

Comment:
Hmm seems like this can be refactored out so that we don't use an accessor unnecessarily. But nevermind, we leave that for v1.4. Can we add a TODO?

```suggestion

    // TODO: Refactor and remove accessor
    public Tag getTag() {
        return this.tag;
    }
```